CONTRIBUTING.md

```diff
@@ -0,0 +1,25 @@
+# Contributing to lc0
+
+There are several project guidelines we prefer that contributors stick to.
+
+This codebase uses Semantic Versioning. This codebase uses Semantic Versioning. A release is the final commit for that version number, and all subsequent commits are development for the next version. Updating the version (i.e. making a release) should be done only with the automated tool for it, `scripts/bumpversion.py`. Any commit that changes the version must consist solely of that version change; any commit which changes anything but the version cannot also change the version.
+
+`master` is the default branch, and the active development branch. `release` will always point to the latest tag. `master` (when it differs from `release`, i.e. almost always) will have a version which includes a postfix attached to the patch section of the version. `release` does not have a postfix attached to the version.
+
```

Comment:
"""
### Allowed features

The Lc0 is still in early stages of development, and not yet reached the point where we are ready to add small tweaks to add few points of a rating. Large code changes still happen, and having lots of small optimizations when one has to keep slows process down.

Therefore, as a rule, search algorithm tweaks that give gain of less than ~30 Elo points are discouraged at this point.
(this limit will gradually be lowered as Lc0 code matures, eventually to 0.0 Elo).

#### Adding new command line flags/ uci parameters

Only add new parameters if users can significantly (>30 Elo) benefit by tweaking it. We don't want to make every single constant configurable (or rather, users don't want to see hundreds of parameters which don't do anything).

Try to minimize number of parameters that your feature introduces. If your feature introduces several parameters, every individual parameter should be significant (i.e. tweaking it with others fixes will give >30 Elo).

#### Adding feature for testing

It is fine to temporary commit a feature of unknown Elo gain to that people could test it. It's also fine to expose many parameters for the feature initially so that people could tune them. However, if the tweak didn't prove to be significant, it should be removed after few weeks.
Same for parameters: it's fine to have them initially for tuning, but if gain of tweaking particular parameter is minuscule, that parameter should be removed.

"""

Reply:
Also about command line flags:

"""
Every parameter consists of command line flag and description.
Description is used as:
- Help text when `lc0` is run with `--help` parameter
- Uci option name (for parameters used in `uci` mode).

Therefore description should:
- Not end with a period (UCI options are not ended with a dot, and dot is added automatically when outputting help)
- Be short (otherwise UCI parameter won't fit in chess UI)
- Be different from command line flag (otherwise it looks silly in `--help`). In should explain command line flag instead
- It should be clear what it means as UCI parameter.

"""